components/shared/profile/edit/EditProfile.tsx

```diff
@@ -0,0 +1,229 @@
+"use client";
+import { IUser } from "@/database/models/user.model";
+import React, { useEffect } from "react";
+import { usePathname } from "next/navigation";
+import { updateUser } from "@/lib/actions/user.actions";
+import { useForm, useFieldArray, SubmitHandler } from "react-hook-form";
+
+import * as z from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import CustomButton from "../../CustomButton";
+import { Input } from "@/components/ui/input";
+import LearningGoal from "../LearningGoal";
+
+import ExperienceLevels from "../ExperienceLevels";
+
+import plusCircle from "@/public/icons/pluscircle.svg";
+import {
+  AvailabilitySchema,
+  CompleteProfileEditSchema,
+} from "@/lib/profileValidations";
+import { PlusSquareIcon } from "lucide-react";
+import Image from "next/image";
+import { Button } from "@/components/ui/button";
+
+interface EditProfileProps {
+  user?: Partial<IUser>;
+  _id?: string;
+}
+type Goal = {
+  name: string;
+  completed: boolean;
+};
+
+const EditProfile = ({ user }: EditProfileProps) => {
+  const pathname = usePathname();
+  const dbGoalNames = user?.learningGoals?.map((goal, idx) => ({
+    name: goal.name,
+    key: idx,
+    completed: goal.completed,
+  }));
+  const dbGoalCompleted = user?.learningGoals?.map((goal) => ({
+    name: goal.completed === true ? goal.name : "",
+    completed: goal.completed === true ? goal.name : false,
+  }));
+  const {
+    register,
+    handleSubmit,
+    control,
+    setValue,
+    setFocus,
+    watch,
+    formState: { errors },
+  } = useForm<z.infer<typeof CompleteProfileEditSchema>>({
+    resolver: zodResolver(CompleteProfileEditSchema),
+    defaultValues: {
+      email: user?.email || "",
+      fullname: user?.fullname || "",
+      location: user?.location || "",
+      portfolio: user?.portfolio || "",
+      learningGoals: dbGoalNames || [],
+      technologies: user?.technologies || [],
+      availability: user?.availability || [],
+      socials: user?.socials || [],
+      experiences: user?.experiences?.map((experience) => experience) || [], // Added 'experienceLevel' property
```

Comment:
```suggestion
      experiences: user?.experiences.map((experience) => ({
        name: experience,
      })),
```

This is backwards. Remember we want to take the Array of strings from the Database and convert them into an object for `useFieldArray`.